src/order/bounded_order.lean

```diff
@@ -1737,6 +1762,61 @@ instance : complemented_lattice αᵒᵈ :=
 
 end complemented_lattice
 
+-- TODO: Define as a sublattice?
```

Comment:
I'm still not quite convinced that the predicate + subtype approach is the best way to go. Although dot notation is convenient, won't it cause a lot of annoyance transporting between `is_complemented` and `complementeds`, as it is already for `is_unit`?

With this setup, we definitely need some explicit maps between `complementeds` and `is_complemented`, along the lines of what we have for `units` and `is_unit`.

Reply:
We now have quite a lot of experience with `is_upper_set`/`upper_set`, and it seems everything works smoothly. Of course there's some "morally duplicated" code, but IMO it's API making every use ever so less irritating.